code/modules/tgui_input/say_modal/tgui_say_typing.dm

```diff
@@ -0,0 +1,50 @@
+/mob/set_stat(new_stat)
+	. = ..()
+	if(.)
+		set_typing_indicator(FALSE)
+
+/mob/Logout()
+	set_typing_indicator(FALSE)
+	return ..()
```

Comment:
IMO it would be better to add these to their parent procs, or at least the logout one, since `mob/Logout` has its own file entirely for that proc.